node/rest/src/routes.rs

```diff
@@ -334,6 +334,13 @@ impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
             return Err(RestError(format!("Unable to broadcast transaction '{}' (node is syncing)", fmt_id(tx.id()))));
         }
 
+        // If the transaction exceeds the transaction size limit, return an error.
+        // TODO: Should this be a blocking task?
+        let buffer = Vec::with_capacity(N::MAX_TRANSACTION_SIZE);
```

Comment:
Normally `with_capacity` is a good idea. However, here the consequence of this is 128KB being allocated for each transaction call. At `30 TPS` this is roughly `~225 MB` worth of memory allocation per minute. 

Most transactions won't be that large so it's a lot of allocation.

Given credits.aleo functions are likely to be the largest we see. Maybe here's better to allocate the size of something like `transfer_public` to start?

Reply:
+1 to this. The max size should not be allocated each time

Reply:
`transfer_public` is `2880 bytes` so we could bound this by `3000`.

Reply:
Set to `3000` in [2b75824](https://github.com/AleoHQ/snarkOS/pull/3229/commits/2b75824d45a8d985507dfceba72077796366738a).